### DISCUSSION TITLE
The Catmull slider curve type has disappeared from the curve type dropdown list

### DISCUSSION THREAD

**Discussion by Digitalfear117** [ppy/osu#27714] [Ideas]
### Type

Other

### Bug description

In previous versions of osu! lazer up until 2023.1130.0, it was possible to set the slider curve type to Catmull, but in https://github.com/ppy/osu/pull/25409 the ability to add them was removed. It seems that in all instances where you could previously create a Catmull curve, the B-spline type has replaced it, rather than being an additional option. Now it is only possible to choose the Catmull curve type on an anchor if the head of the slider was already previously set to the Catmull curve type. However, since the Catmull type is not exposed to the user when creating a slider, it is not possible to create any additional Catmull sliders curves without having one previously available within the map. (You would need to copy and paste an existing Catmull slider to place a new one).

It is entirely possible that this was changed on purpose, but even in the changelog for 2023.1130.0 you can still see the Catmull type being displayed within the editor screenshots, leading me to believe this could have been a mistake. Here is the screenshot seen in that changelog. 
![image seen in the changelog showing the catmull option](https://github.com/ppy/osu/assets/28713262/aeba066c-1ce6-4934-832b-acbc24dafcd2)

Here is how it looks in the most recent version of lazer ![screenshot of the curve types available in the most recent version of lazer, with the catmull option being notably missing](https://github.com/ppy/osu/assets/28713262/96c63aeb-cbaa-4d39-be70-72659913c6f6)

This was noticed by my friend [Celerrin](https://osu.ppy.sh/users/4873016) as we had collaborated on making a 2008 style beatmap together, and he was using lazer to do his parts of the collab. However after updating lazer his ability to create Catmull sliders was removed. He had to make the choice of either constantly downgrading lazer, or switching to an older version of the stable client. These were his only options to make the Catmull slider shapes we wanted for the map. In the end, it was easier to just use an older stable build, which is a shame because the lazer editor has a lot of cool ideas and potential.

If this was an intentional change I think it should be reconsidered. Quite a few maps in recent years have been ranked with Catmull sliders. This is despite the fact that this object type isn't selectable in the modern stable editor, and has to be added in with .osu editing. Instead of doing that, I think it would be better to have it available to the user like in previous versions of lazer. Here is a list of every ranked or loved map that I had in my beatmap collection that was ranked within the past 8 years or so. These examples show that even after many years of the Catmull slider type being hidden from users in stable, mappers are still showing an active interest in this slider curve type!

| Beatmapset ID | Beatmap ID | Difficulty Name | Game Mode | # of Catmulls | Status |
| ---------------- | ------------- | ----------------- | -------------- | -------------- | ------- |
| 1074744 | 2249030 | Flurry of Feelings | osu | 12 | Ranked |
| 1093465 | 2285424 | ExpertPlus | osu | 13 | Ranked |
| 1219288 | 2593763 | LASSE'S EXPERT | osu | 3 | Ranked |
| 1237363 | 2571858 | Aspire | osu | 1 | Loved |
| 1241389 | 2580772 | Intermedio | osu | 5 | Ranked |
| 1518426 | 3332141 | Breezin' | osu | 2 | Ranked |
| 1518426 | 3332142 | Cruisin' | osu | 3 | Ranked |
| 1518426 | 3108227 | Hard ROCK! | osu | 5 | Ranked |
| 1518426 | 3108228 | Sweatin' | osu | 5 | Ranked |
| 1545821 | 3323223 | WashedUp's Insane (feat. 1103) | osu | 1 | Ranked |
| 1573985 | 3267112 | riffy's hard | osu | 13 | Ranked |
| 1588934 | 3346859 | Apollo's Hard | osu | 1 | Ranked |
| 1615650 | 3303190 | Hard | osu | 5 | Ranked |
| 1615650 | 3298592 | Luscent's Insane | osu | 8 | Ranked |
| 1621130 | 3309941 | Lesjuh's Aye!! | osu | 2 | Ranked |
| 1643400 | 3364766 | Bloxi's Normal | osu | 80 | Ranked |
| 1653563 | 3376211 | Easy | osu | 3 | Ranked |
| 1677082 | 3426496 | 2022 Will Be Our Year. | osu | 2 | Ranked |
| 1677082 | 3426498 | Normal | osu | 2 | Ranked |
| 1738470 | 3553225 | Hard ROCK! | osu | 6 | Ranked |
| 1738470 | 3553235 | Sweatin' | osu | 6 | Ranked |
| 1758567 | 3598905 | hard | osu | 2 | Ranked |
| 1800845 | 3692015 | FOOTPRINTS | osu | 162 | Ranked |
| 1800845 | 4442924 | BEGINNER | osu | 91 | Ranked |
| 1800845 | 4523345 | EASY | osu | 119 | Ranked |
| 18315 | 69131 | ignore's Another | osu | 2 | Loved |
| 1918826 | 3959698 | Insane (feat. riffy) | osu | 1 | Ranked |
| 1923862 | 3971723 | Easy | osu | 1 | Ranked |
| 2068931 | 4328856 | Normal | osu | 1 | Ranked |
| 217667 | 537937 | Aquarium Easy | osu | 35 | Loved |
| 217667 | 537938 | Aquarium Hard | osu | 55 | Loved |
| 31297 | 102921 | CTB newer competition | catch | 10 | Loved |
| 31297 | 102923 | CTB pro competition | catch | 10 | Loved |
| 333139 | 768942 | tyui's Extra | osu | 6 | Ranked |
| 668888 | 1429741 | Cheesecake's Normal. | osu | 1 | Ranked |
| 864679 | 1808290 | EX | osu | 1 | Ranked |
| 864679 | 1946522 | MX | osu | 2 | Ranked |
| 88180 | 462224 | Collab Normal | osu | 8 | Ranked |
| 902215 | 2131562 | INSANE feat. EM0NKY | osu | 2 | Ranked |
| 903393 | 1885883 | Cowboy in Outer Space | osu | 1 | Ranked |

### Version

2023.1130.0 and onward

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14735537/compressed-logs.zip)

**Comment by peppy** [comment]
It was intentionally removed.

**Reply by Digitalfear117** [reply]
Gotcha, I understand that the Catmull curve type is a bit outdated, but it would be very cool if they could be usable somehow, even if it’s tucked behind a setting or something. Maybe something like `Show legacy curve type` in the setup section. If this radio box is enabled, the Catmull type would show up in the dropdown again, maybe even with a deprecated notice after the name.

![Mockup of what a proposed Show legacy curve type could look like in the setup section of the editor](https://github.com/ppy/osu/assets/28713262/237ccf12-5d53-484c-a390-9c98bfc23436)

Here's an alternative mockup of what it could look like if it was in the main settings panel as well. I also threw in a couple of ones from stable too.
![Alternative mockup of what it could look like in the main settings panel, with other options seen in stable](https://github.com/ppy/osu/assets/28713262/6bebd849-8a5f-4db0-9908-039fc49f8046)


![Mockup of what a the curve type dropdown list could look like with Catmull and an added deprecated notice after it](https://github.com/ppy/osu/assets/28713262/ef69082f-c4be-4fcc-9d9b-f581e78cc1b2)

I mean, its a free curve type, it's already added to the game for compatibility, why not let users go crazy and use them if they want? :D

**Reply by velamyGH** [reply]
kind of agree, id like to see it return in some way if possible. maybe under the design section in the setup tab if there really is a concern with new players using it for some reason
